### PR TITLE
add REST APIs to directory

### PR DESCRIPTION
**Pull request by gertd**

### CODE REVIEW

**Comment by carabasdaniel** [comment]
I encountered the following issue when generating a dev build:
```
>>> executing buf generate ../pb-directory/bin/directory.bin
Failure: plugin grpc-gateway: must not set request body when http method is DELETE except allow_delete_body option is true: DeleteObjectType
```

**Comment by gertd** [comment]
> I encountered the following issue when generating a dev build:
> 
> ```
> >>> executing buf generate ../pb-directory/bin/directory.bin
> Failure: plugin grpc-gateway: must not set request body when http method is DELETE except allow_delete_body option is true: DeleteObjectType
> ```

You need an updated buf.gen.yaml in go-directory which contains:
```
version: v1

plugins:
  - name: go
    out: .
    opt: 
      - paths=source_relative
  - name: go-grpc
    out: .
    opt: 
      - paths=source_relative
      - require_unimplemented_servers=false
  - name: grpc-gateway
    out: .
    opt: 
      - paths=source_relative
      - logtostderr=true
      - generate_unbound_methods=true
      - allow_delete_body=true
  - plugin: openapiv2
    out: ./openapi
    opt:
      - allow_delete_body=true
```

File: proto/aserto/directory/reader/v2/reader.proto [hunk 1]
```diff
@@ -4,36 +4,366 @@ package aserto.directory.reader.v2;
...
+                }
+            }
+        };    
+    };
     
     // object methods
-    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {};
+    rpc GetObject(GetObjectRequest) returns (GetObjectResponse) {
+        option (google.api.http) = {
+            post: "/api/v2/directory/object"
```
Comment (ronenh) [inline proto/aserto/directory/reader/v2/reader.proto:197]: I know that some Get requests have complex structure with optional inputs and need to be exposed as POST endpoints, but for simple ones like this, would it make sense to expose it as: 
`get: "/api/v2/directory/objects/{type}/{key}`

And similarly in the writer we'd have `put: "/api/v2/directory/objects/{type}/{key}` and the same for delete.

File: proto/aserto/directory/reader/v2/reader.proto [hunk 2]
```diff
@@ -4,36 +4,366 @@ package aserto.directory.reader.v2;
...
+            body: "*"
+        };
+        option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
```
Comment (gertd) [inline proto/aserto/directory/reader/v2/reader.proto:200]: The current usage of optional fields inside ObjectIdentifier prohibits the usage of them as path parameters

```
        option (google.api.http) = {
            post: "/api/v2/directory/object"
            body: "*"
            additional_bindings: [
            {
                get: "/api/v2/directory/object/{param.type}/{param.key}"
            }
            ]
        };
```

will result in 
```
>>> executing buf generate ../pb-directory/bin/directory.bin
Failure: plugin grpc-gateway: optional field not allowed in field path: type in param.type
Error: running "/Users/gertd/workspace/src/github.com/aserto-dev/go-directory/.ext/gobin/buf-v1.9.0/buf generate ../pb-directory/bin/directory.bin" failed with exit code 1
```

**Review by ronenh — APPROVED** [review]

**Pull request merged**